### PR TITLE
fix: app crash and workaround for image in rich push

### PR DESCRIPTION
**Pull request by ami-aman**
closes https://github.com/customerio/issues/issues/8112

This PR also contains workaround fix for RH not showing image in rich push notification.

### CODE REVIEW

File: Remote Habits/Manager/DataManager.swift [hunk 1]
```diff
@@ -76,6 +76,7 @@ class HabitDataManager: CoreDataManager {
                     objectUpdate.setValue(endTimeValue.toDate(withFormat: .time12Hour), forKey: "endTime")
                 }
             }
+            managedContext.mergePolicy = NSOverwriteMergePolicy
```
Comment (jatinn) [inline Remote Habits/Manager/DataManager.swift:79]: maybe a comment as to why this fixes the issues

File: Notification Service/NotificationService.swift [hunk 2]
```diff
@@ -11,6 +12,7 @@ class NotificationService: UNNotificationServiceExtension {
         // If you use more service then Customer.io for sending rich push messages,
         // you can check if the SDK handled the rich push for you. If it did not, you
         // know that the push was *not* sent by Customer.io and you can try another way.
+        CustomerIO.initialize(siteId: Env.customerIOSiteId, apiKey: Env.customerIOApiKey, region: Region.US)
```
Comment (jatinn) [inline Notification Service/NotificationService.swift:15]: probably fine for now but we should likely be saving the region into env as well

**Review by jatinn — APPROVED** [review]

File: Remote Habits/Manager/DataManager.swift (hunk 1 above)
Comment (ami-aman) [reply Remote Habits/Manager/DataManager.swift:79]: ✅

File: Notification Service/NotificationService.swift (hunk 2 above)
Comment (ami-aman) [reply Notification Service/NotificationService.swift:15]: Yes, this is a temporary workaround and will be removing it once we have the changes up in the SDK. Probably I should add a comment here.

Comment (ami-aman) [reply Notification Service/NotificationService.swift]: Added a comment mentioning the same.

**Pull request merged**